src/quadrature/quadrature_simpson_3D.C

```diff
@@ -131,6 +131,95 @@ void QSimpson::init_3D(const ElemType, unsigned int)
       }
 
 
+      //---------------------------------------------
+      // Pyramid quadrature rules
+    case PYRAMID5:
+    case PYRAMID13:
+    case PYRAMID14:
+      {
+        _points.resize(14);
+        _weights.resize(14);
+
+        _points[0](0) = -1.;
+        _points[0](1) = -1.;
+        _points[0](2) = 0.;
+
+        _points[1](0) = 1.;
+        _points[1](1) = -1.;
+        _points[1](2) = 0.;
+
+        _points[2](0) = 1.;
+        _points[2](1) = 1.;
+        _points[2](2) = 0.;
+
+        _points[3](0) = -1.;
+        _points[3](1) = 1.;
+        _points[3](2) = 0.;
+
+        _points[4](0) = 0.;
+        _points[4](1) = 0.;
+        _points[4](2) = 1.;
+
+        _points[5](0) = 0.;
+        _points[5](1) = -1.;
+        _points[5](2) = 0.;
+
+        _points[6](0) = 1.;
+        _points[6](1) = 0.;
+        _points[6](2) = 0.;
+
+        _points[7](0) = 0.;
+        _points[7](1) = 1.;
+        _points[7](2) = 0.;
+
+        _points[8](0) = -1.;
+        _points[8](1) = 0.;
+        _points[8](2) = 0.;
+
+        _points[9](0) = 0.;
+        _points[9](1) = -0.5;
+        _points[9](2) = 0.5;
+
+        _points[10](0) = 0.5;
+        _points[10](1) = 0.;
+        _points[10](2) = 0.5;
+
+        _points[11](0) = 0.;
+        _points[11](1) = 0.5;
+        _points[11](2) = 0.5;
+
+        _points[12](0) = -0.5;
+        _points[12](1) = 0.;
+        _points[12](2) = 0.5;
+
+        _points[13](0) = 0.;
+        _points[13](1) = 0.;
+        _points[13](2) = 0.;
+
+        // These are of dubious value since we can't integrate on the
```

Comment:
I just worry that anyone accidentally using this quadrature rule for integration will now get an extremely singular mass matrix (and other unexpected results) rather than an error message... is there no way we can warn about this somehow?

Reply:
Hmm... I guess they will get an error when they `reinit()` due to `JxW` being zero/undefined at the apex node, so maybe that's good enough.

Reply:
That's what I'm thinking.  And then if we ever *do* figure out how to ever avoid the undefined JxW (a special case for pyramids where the mapping function turns out to be linear and so the map derivatives at the apex are consistent??) then this will actually be useful for node-lumped integration too.